openadapt/ml_models/ml_models.py

```diff
@@ -0,0 +1,58 @@
+from openadapt import models
+import openai_module
+from openai_module.gpt3_davinci import infer, finetune, MODALITIES
+from openadapt.crud import (
+    get_recording_by_id,
+    get_latest_recording,
+    get_window_events,
+    get_action_events,
+)
+from loguru import logger 
+
+
+class Modality:
+    TEXT = "TEXT"
+    IMAGE = "IMAGE"
+
+class License:
+    COMMERCIAL_USE = "COMMERCIAL_USE"
+    NON_COMMERCIAL_USE = "NON_COMMERCIAL_USE"
+
+class Capability:
+    TRAINING = "TRAINING"
+    TUNING = "TUNING"
+    USABLE_OUTPUT = "USABLE_OUTPUT"
+    INFERENCE = "INFERENCE"
+
+MODALITY_BY_DB_MODEL = {
+    models.ActionEvent: Modality.TEXT,
+    models.Screenshot: Modality.IMAGE,
+    models.WindowEvent: Modality.IMAGE,
+}
+
+ML_MODELS = [
+    {
+        "name": "gpt3_davinci",
+        "provider": openai_module.gpt3_davinci,
+        "finetune": finetune,
+        "infer": infer,
+        "modalities": MODALITIES,
+        "context_length": 4096,
+        "license": License.COMMERCIAL_USE,
+        "capability": [Capability.TRAINING, Capability.TUNING, Capability.USABLE_OUTPUT, Capability.INFERENCE]
+    }
+]
+
```

Comment:
As I mention below, I would just float the thoughts:

Should we break down this into `pydantic` class or `dataclass`  as well ?

Reply:
added a class ```Availability``` to address physical location 👍